framework/routing/__init__.py

```diff
@@ -187,15 +197,25 @@ def render_jinja_string(tpl, data):
     pass
 
 mako_cache = {}
-def render_mako_string(tpldir, tplname, data):
+def render_mako_string(tpldir, tplname, data, safe=False):
+    """Render a mako template. Optional safe argument to activate markup escaping"""
+    if safe is True:
```

Comment:
Could this conditional be replaced with `lookup_obj = _tpl_lookup_safe if safe else _tpl_lookup`?

I don't see how `default_filters` is useful later. Passing `lookup_obj` to the `Template` instantiation should set the filters. If it doesn't, then they can be accessed directly at `lookup_obj['template_args']['default_filters']`.